python/dglke/tests/test_dataset.py

```diff
@@ -0,0 +1,243 @@
+# -*- coding: utf-8 -*-
+#
+# setup.py
+#
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os, unittest
+from dataloader import get_dataset
+
+def gen_udd_files(extra_train=None, extra_valid=None, extra_test=None):
+    train_f = 'train.tsv'
+    valid_f = 'valid.tsv'
+    test_f = 'test.tsv'
+    emap = 'entity.dict'
+    rmap = 'relation.dict'
+    with open(train_f, 'w+') as f:
+        f.write('1\t0\t0\n')
+        f.write('2\t0\t0\n')
+        f.write('3\t0\t4\n')
+        f.write('4\t0\t0\n')
+        if extra_train is not None:
+            for l in extra_train:
+                f.write(l)
+
+    with open(valid_f, 'w+') as f:
+        f.write('2\t0\t1\n')
+        if extra_valid is not None:
+            for l in extra_valid:
+                f.write(l)
+
+    with open(test_f, 'w+') as f:
+        f.write('2\t0\t3\n')
+        if extra_test is not None:
+            for l in extra_test:
+                f.write(l)
+
+    with open(emap, 'w+') as f:
+        f.write('a\t0\n')
+        f.write('b\t1\n')
+        f.write('c\t2\n')
+        f.write('d\t3\n')
+        f.write('d\t4\n')
+
+    with open(rmap, 'w+') as f:
+        f.write('A\t0\n')
+
+    return [emap, rmap, train_f, valid_f, test_f]
+
+def cleanup(files):
+    for f in files:
+        os.remove(f)
+
+class TestUDDDataset(unittest.TestCase):
+    def test_udd_noint_triplets(self):
```

Comment:
could you write comments to describe what error the test tries to check?

Reply:
OK